### DISCUSSION TITLE
How does one implement a join or connect or set in a self relational table for MongoDB

### DISCUSSION THREAD

**Discussion by FentonA** [prisma/prisma#12833] [Q&A] [answered]
Running into an error when attempting to connect two relational data (from self relational model) with MongoDB. For the below model, how should go about joining two users through the 'connect' or 'set' feature? Any help of any kind would be greatly appreciated
Schema model: 
```
model User {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  email       String   @unique
  description String?
  name        String?
  username    String?    
  password    String?
  followedBy  User[] @relation("UserFollows", fields:[followedByIDs], references: [id])
  followedByIDs String[] @db.ObjectId
  following   User[] @relation("UserFollows", fields: [followingIDs], references: [id] )
  followingIDs String[] @db.ObjectId
  role      Role     @default(USER)
  reivews   Review[]
  emailVerified DateTime? @map("email_verified")
  image         String?
  accounts      Account[]
  sessions      Session[]

  @@map("users")
  @@fulltext([name(sort:Desc), username, email])
}
```

Prisms client implementation: 

```
        await prisma.user.update({
            where:{
                email:'alf@votaryfilms.com'
            },
            data:{
                following: {
                    connect:{
                        id: data
                    }
                }
            }
        })
```

Error:
`The current database provider doesn't support a feature that the query used: Unhandled and unsupported value mapping for MongoDB: 6223c13a26bb3d44fc98dad5 as Int.`

**Comment by austincrim** [comment, marked as answer]
Discussion happening [here](https://github.com/prisma/prisma/discussions/12839)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
